app/data_model/database.py

```diff
@@ -68,8 +71,32 @@ def create_session_and_engine():
         logger.exception(e)
         raise e
 
-
 db_session, engine = create_session_and_engine()
 base.query = db_session.query_property()
 
 base.metadata.create_all(engine)
+
+
+@event.listens_for(engine, "engine_connect")
+def test_connection(connection, branch):
+    if branch:
+        # "branch" refers to a sub-connection of a connection,
+        # we don't want to bother testing on these.
+        return
```

Comment:
What does this return? If the function has finished perhaps return None or True? Further on the code returns errors, so be good to have a return value under any scenario.